app_dart/lib/src/request_handlers/refresh_github_commits.dart

```diff
@@ -59,22 +59,16 @@ class RefreshGithubCommits extends ApiRequestHandler<Body> {
     final DatastoreService datastore = datastoreProvider(config.db);
 
     for (String branch in await config.flutterBranches) {
-      // TODO(keyonghan): use time of latest commit for [hours] in github API query, https://github.com/flutter/flutter/issues/54284
-      int hours = config.existingBranchHours;
-
-      if (branch != 'master') {
-        final Set<Commit> commits = await datastore
-            .queryRecentCommits(limit: 1, branch: branch)
-            .toSet();
-
-        /// Set [hours] to be longer if a new branch is found.
-        if (commits.isEmpty) {
-          hours = config.newBranchHours;
-        }
+      final List<Commit> lastCommit =
+          await datastore.queryRecentCommits(limit: 1, branch: branch).toList();
```

Comment:
queryRecentCommits does not provide all the context of what the method is doing, should we rename it to queryNotProcessedCommits?

Reply:
The returned results are commits that have been processed. Other APIs are calling `queryRecentCommits` as well.

To me, it does what it suggests: query the recent commits in datastore filtered by the `limit` number and the `branch`.

Here it returns the most recent single (processed) commit in datastore, which is expected.

Please let me know if I understand incorrectly.

Reply:
queryRecentCommits is just grabbing all the commits on datastore and returning the one on the top when ordering by timestamp. That means that we are potentially processing the same commit several times. I'd suggest to improve the query to return only commits that haven't been processed yet.

Reply:
I think there is some misunderstanding here.

We are trying to get the latest commit that has been processed in datastore, rather than get the commits that haven't been processed. All commits in datastore are already processed.

Then we use this commit's timestamp as the benchmark to call Github API to return newer commits.